From 249f0b75cf263f8ecdd181fd1a8babfa545dbe01 Mon Sep 17 00:00:00 2001
Message-Id: <patch-23.24-249f0b75cf2-20220309T121722Z-avarab@gmail.com>
In-Reply-To: <cover-00.24-00000000000-20220309T121722Z-avarab@gmail.com>
References: <cover-00.24-00000000000-20220309T121722Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 30 Nov 2021 12:36:52 +0100
Subject: [PATCH 23/24] revisions API: have release_revisions() release
 "date_mode"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Extend the the release_revisions() function so that it frees the
"date_mode" in the "struct ref_info".

This uses the date_mode_release() function added in 974c919d36d (date
API: add and use a date_mode_release(), 2022-02-16). As that commit
notes "t7004-tag.sh" tests for the leaks that are being fixed
here. That test now fails "only" 44 tests, instead of the 46 it failed
before this change.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 revision.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/revision.c b/revision.c
index 81f4a36fc9d..9d03e7805ff 100644
--- a/revision.c
+++ b/revision.c
@@ -2954,6 +2954,7 @@ void release_revisions(struct rev_info *revs)
 	release_revisions_cmdline(&revs->cmdline);
 	object_array_clear(&revs->boundary_commits);
 	clear_pathspec(&revs->prune_data);
+	date_mode_release(&revs->date_mode);
 	release_revisions_mailmap(revs->mailmap);
 	free_grep_patterns(&revs->grep_filter);
 	/* TODO (need to handle "no_free"): diff_free(&revs->diffopt) */
-- 
2.35.1.1295.g6b025d3e231

